modules/centreon-stream-connectors-lib/sc_metrics.lua

```diff
@@ -280,4 +311,4 @@ function ScMetrics:build_metric(format_metric)
   end
 end
 
```

Comment:
```suggestion
--- get_full_metric_name: build the metric_name including metric instance and subinstances if there are any
-- @param metric_info (table) a table with all the metric information (unit, name, instances...)
-- @return metric_name (string) the full name of the metric including instances and subinstances if there are any
function ScMetrics:get_full_metric_name(metric_info)
  local metric_name = metric_info.metric_name

  if metric_info.instance ~= "" then
    if #metric_info.subinstance ~= 0 then
      metric_name = metric_info.instance .. '~' .. table.concat(metric_info.subinstance, '~') .. '#' .. metric_info.metric_name
    else
      metric_name = metric_info.instance .. '#' .. metric_info.metric_name
    end
  end

  return metric_name
end

```

if such function is created, it needs to be documented in the sc_metrics.md add readme files